src/abr/pensieve.cc

```diff
@@ -0,0 +1,105 @@
+#include "pensieve.hh"
+#include "ws_client.hh"
+
+using namespace std;
+
+//TODO: Reference already existing put_field()
```

Comment:
Idea for this TODO: we can add two files such as `serialization.hh` and `serialization.cc` in `src/net`, and make every file using `put_field`/`get_uint16` reference to `serialization.hh`.